packages/kbn-config-schema/src/types/string_type.ts

```diff
@@ -37,24 +39,29 @@ export class StringType extends Type<string> {
             }
             return value;
           });
+
     if (options.minLength !== undefined) {
-      schema = schema.custom(
-        convertValidationFunction((value) => {
-          if (value.length < options.minLength!) {
-            return `value has length [${value.length}] but it must have a minimum length of [${options.minLength}].`;
-          }
-        })
-      );
+      schema = schema
+        .custom(
+          convertValidationFunction((value) => {
+            if (value.length < options.minLength!) {
+              return `value has length [${value.length}] but it must have a minimum length of [${options.minLength}].`;
+            }
+          })
+        )
+        .meta({ [META_FIELD_X_OAS_MIN_LENGTH]: options.minLength });
```

Comment:
Due to our custom handling of string length validation, these values are currently lost. In order to avoid changing behaviour we use the `meta` function ([docs](https://joi.dev/api/?v=17.12.3#anymetameta)) to attach a new piece of metadata containing string min/max length will propagated to through `Joi.schema.describe` ([docs](https://joi.dev/api/?v=17.12.3#anydescribe))